src/panels/lovelace/cards/hui-thermostat-card.ts

```diff
@@ -0,0 +1,309 @@
+import { html, LitElement } from "@polymer/lit-element";
+import { classMap } from "lit-html/directives/classMap.js";
+import roundSliderStyle from "round-slider/dist/roundslider.min.css";
+
+import "../../../components/ha-card.js";
+import "../../../components/ha-icon.js";
+import "../../../resources/jquery.roundslider";
+
+import { HomeAssistant } from "../../../types.js";
+import { HassLocalizeLitMixin } from "../../../mixins/lit-localize-mixin";
+import { LovelaceCard, LovelaceConfig } from "../types.js";
+
+const thermostatConfig = {
+  radius: 150,
+  min: 50,
+  max: 90,
+  step: 1,
+  circleShape: "pie",
+  sliderType: "min-range",
+  value: 70,
+  startAngle: 315,
+  width: 5,
+  lineCap: "round",
+  handleSize: "+10",
+  showTooltip: false,
+};
+
+const modeIcons = {
+  auto: "mdi:autorenew",
+  heat: "hass:fire",
+  cool: "hass:snowflake",
+  off: "mdi:fan-off",
+};
+
+interface Config extends LovelaceConfig {
+  entity: string;
+}
+
+let loaded: Promise<void>;
+
+export class HuiThermostatCard extends HassLocalizeLitMixin(LitElement)
+  implements LovelaceCard {
+  public hass?: HomeAssistant;
+  protected config?: Config;
+  protected roundSlider: any;
```

Comment:
`roundSlider?`

Reply:
I wanted to be able to use the data and function through out the class

Reply:
So ? You're not defining it in the constructor so it should be `?`. Also, instead of `any`, can you type some of the API ?

Just something like this:

```ts
type RoundSlider = {
  setValue: (value: number) => void;
}
```

Reply:
ahh. I saw that as roundSlider ? Like your were questioning why it was there haha. oops

Reply:
my bad.